src/utils/child-pathname.ts

```diff
@@ -0,0 +1,3 @@
+export const isChildPathnameOf = (url: string, parentUrl: string) => {
+  return url === parentUrl || new RegExp(`${url}[^\\w\\d]`, 'g').test(parentUrl);
```

Comment:
The regex allows: https://abc and https://abcd, right?

Reply:
Previously I was using `String.prototype.includes`, which does allow that. Here I use the regex to solve that problem, so suppose we have https://dev.open-paas.org/calendar:

- https://dev.open-paas.org/calendarasdfsdf is invalid
- https://dev.open-paas.org/calendar/something is valid
- https://dev.open-paas.org/calendar?something=another_thing is valid

Reply:
Updated the regex to disallow - _ . ~ also.

Reply:
These can be found at https://tools.ietf.org/html/rfc3986#section-2.